Limpeza do Código

SofiaSFaber opened the pull request:
A limpeza do código levou em consideração comentários, otimizações, renomeações e reestruturações.

Primeiramente, todos os comentários foram removidos para ficar mais fácil de mexer no fonte. Muito dos comentários eram óbvios e repetições do que o código estava claramente fazendo. Depois de todas as outras alterações de limpeza, foram adicionados alguns comentários em relação a lógica de busca e manipulação de dados que pode ser um pouco mais difícil de entender.

Sobre a estruturação, o código foi separado em algumas funções com responsabilidades mais específicas como, ler um CSV, pegar os dados do CSV, fazer a busca no banco de dados e criar um JSON. Também foram modificados os espaçamentos, tanto verticais como horizontais, para facilitar a leitura do código.

De otimização, a mudança principal foi na função GetNCBIData, já que ela possuía um grande número de linhas e um "if" gigante. Também foram removidos alguns prints que eram mais de controle de fluxo e não tão necessários. Como as outras funções estavam bem objetivas e tinham poucas manipulações, não foi necessário mais otimizações.

Por último, as renomeações foram feitas para deixar o código o mais claro possível, onde com os nomes das funções e das variáveis é possível entender o que ele tem como funcionalidade. Também, as renomeações seguiram um padrão em inglês para o fonte ser acessível a mais pessoas.

SofiaSFaber committed:
Separação do código em funções

SofiaSFaber committed:
Remoção dos comentários

SofiaSFaber committed:
Renomeações

SofiaSFaber committed:
Remoção de prints desnecessários

SofiaSFaber committed:
Alteração de espaçamentos

SofiaSFaber committed:
Otimização da função GetNCBIData e algumas renomeações

SofiaSFaber committed:
Adiciona comentarios essenciais
Com as renomeaçoes e organizaçao do codigo, grande parte dos comentarios ja não era mais necessaria.
Porem, alguns detalhes em relaçao a logica da busca e da manipulaçao de dados podem nao ficar tao claros,
e por isso foram adicionados breves comentarios para explicar essa parte.

aliceronascimento commented on trabalhobanco.py:
```diff
@@ -7,86 +7,89 @@
+  results = {}
+  remove = []
+  
+  for i in range(len(genes)):
+    handle = Entrez.esearch(db = "nucleotide", term = genes[i]) #Pesquisa os genes no banco de dados de nucleotídeos e salva os resultados
+    record = Entrez.read(handle)
+    if len(record['IdList'])>2: #Pega os genes com pelo menos 3 resultados
+      results[genes[i]] = record["IdList"][0], record['IdList'][1], record['IdList'][2]
+    else:
+      remove.append(genes[i])
+  
+  for i in range(len(remove)):
+    genes.remove(remove[i])
+
+  data = {}
+  content = ["GBSeq_references", "GBSeq_strandedness", "GBSeq_locus", "GBSeq_length", "GBSeq_moltype", "GBSeq_topology",
+          "GBSeq_division", "GBSeq_definition", "GBSeq_accession-version"]
+
+  for i in range(len(genes)):
+      handle = Entrez.efetch(db = "nucleotide", id = results[genes[i]][0], retmode = "xml") #Pega os 3 primeiros resultados do NCBI para o gene
+      records1 = Entrez.read(handle)
+      handle = Entrez.efetch(db = "nucleotide", id = results[genes[i]][1], retmode = "xml")
+      records2 = Entrez.read(handle)
+      handle = Entrez.efetch(db = "nucleotide", id = results[genes[i]][2], retmode = "xml")
+      records3 = Entrez.read(handle)
+      for j in range(len(content)):
+        aux = True
+        if (not content[j] in records1[0].keys()) or (not content[j] in records2[0].keys()) or (not content[j] in records3[0].keys()): #Utiliza somente os genes em que os resultados possuem todas as chaves do content
+          aux = False
+      if aux == True:
+        data[genes[i]] = "First Result:", results[genes[i]][0], records1[0]["GBSeq_locus"], records1[0]["GBSeq_length"], records1[0]["GBSeq_strandedness"], records1[0]["GBSeq_moltype"], records1[0]["GBSeq_topology"], records1[0]["GBSeq_division"], records1[0]["GBSeq_definition"], records1[0]["GBSeq_accession-version"], records1[0]["GBSeq_source"], records1[0]["GBSeq_references"],\
+        "Second Result:", results[genes[i]][1], records2[0]["GBSeq_locus"], records2[0]["GBSeq_length"], records2[0]["GBSeq_strandedness"], records2[0]["GBSeq_moltype"], records2[0]["GBSeq_topology"], records2[0]["GBSeq_division"], records2[0]["GBSeq_definition"], records2[0]["GBSeq_accession-version"], records2[0]["GBSeq_source"], records2[0]["GBSeq_references"],\
+        "Third Result:", results[genes[i]][2], records3[0]["GBSeq_locus"], records3[0]["GBSeq_length"], records3[0]["GBSeq_strandedness"], records3[0]["GBSeq_moltype"], records3[0]["GBSeq_topology"], records3[0]["GBSeq_division"], records3[0]["GBSeq_definition"], records3[0]["GBSeq_accession-version"], records3[0]["GBSeq_source"], records3[0]["GBSeq_references"]
+
```
Acho que essa função poderia ser "quebrada" em outras funções menores. Ex: uma função para pegar os 3 primeiros nucleotídeos, outra para pesquisar os genes no banco, remove genes duplicados. Evitaria os comentarios e ficaria mais autoexplicativo

aliceronascimento commented on trabalhobanco.py:
```diff
@@ -7,86 +7,89 @@
+  return mardyCSV
+
+
+def GetCSVGenes(mardyCSV):
```
Acho que alguns comentários ainda são desnecessários, em vista que a função já diz o que ela ta executando

SofiaSFaber committed:
Alterações do que foi comentado pelo revisor do pull request
Separação da função GetNCBIData em funções auxiliares
Remoção de alguns comentários

SofiaSFaber commented:
Realizei as modificações comentadas no commit 12b2347.
Separei a função GetNCBIData com a implementação de 3 funções auxiliares. 
Também removi alguns comentários.

SofiaSFaber committed:
Arruma erro de download
Ao realizar as modificações solicitadas pelo revisor, encontrei um erro cometido
nas renomeações do commit a56eb7323792f4710afbb062f2afaf7140fa646c

aliceronascimento commented:
Ficou muito bom!
